lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -3862,6 +3862,17 @@ sub cleanup_Math {
             push(@texts, $child); } } } }
     # and replace the whole Math with the pieces
     $document->replaceTree([undef, undef, @texts], $mathnode); }
+  # Single token PUNCTs can be taken out of math.
+  elsif (my $xmtok = $document->findnode('ltx:XMath[count(./*) = 1]/ltx:XMTok[@role="PUNCT" or @role="PERIOD"]', $mathnode)) {
```

Comment:
Hmm, I'm wondering if this whole else can be folded into the main if by including it into the xpath.  Something like
```
and not( local-name() = "XMTok" and (@role ="PUNCT" or @role="PERIOD") )
```
within the []. Eliminates a `findnode`, so possibly even faster, depending on the xpath compiler

Reply:
Sounds good, and it seemed a little faster when I benchmarked the article.

I had to also add the check that the PUNCT/PERIOD case is the *only* child in the XMath, since otherwise we match cases that we don't quite want to - in particular in t/math/sampler.tex, the expression
```
\mbox{\tiny $X$},\qquad\mbox{\scriptsize $X$},\qquad\mbox{\normalsize $X$}, %... [snip]
```

Parses as a list, but was wrongly getting its top-level unwrapped to text mode. Actually, I am not sure that is *wrong*, but is not my current intention - something to revisit in the future.